DataFormats/Detectors/TRD/include/DataFormatsTRD/RawData.h

```diff
@@ -214,10 +214,168 @@ struct TRDFeeID {
+        uint32_t nlfilter : 1;
+        uint32_t xtfilter : 1;
+        uint32_t tfilter : 1; 
+        uint32_t gfilter : 1;
+        uint32_t pfilter : 6; 
+    }__attribute__((__packed__));
+  };
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  union {//page 109 section 15.6.4 in tdp
+    uint32_t word3;
+    struct {
+        uint32_t res3 : 6; 
+        uint32_t svnrver : 13;    //readout program svn revision
+        uint32_t svnver  : 13;     //assember programm svn revision
+    }__attribute__((__packed__));
+  };
+};
+
+struct DigitMCMHeader{
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  union {
+    uint32_t word;  //MCM header
+    struct {
+        uint32_t res4 : 4;        // reserve 1100
+        uint32_t eventcount : 20;
+        uint32_t mcm : 4;
+        uint32_t rob : 3;
+        uint32_t yearflag: 1;             //< oct2007 0,  else 1 
+    }__attribute__((__packed__));
+  };
+};
+
+//the odd numbering of 2 3 and 6 are taken from the TDP page 111 section 15.7.3 15.7.4 15.7.5
+struct trdTestPattern2{
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  union {
+    uint32_t word; 
+    struct {
+        uint32_t eventcount : 6; // lower 6 bits of e counter.
+        uint32_t stack: 5;  
+        uint32_t layer : 3;  
+        uint32_t roc : 3;   
+        uint32_t rob: 3;  
+        uint32_t mcmTp2 : 4;
+        uint32_t cpu : 2;  
+        uint32_t counter : 6;
+    }__attribute__((__packed__));
+  };
+};
+struct trdTestPattern3{
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  union {
+    uint32_t word;  
+    struct {
+        uint32_t eventcount: 12;  //lower 12 bits of ecounter
+        uint32_t stack: 5;  
+        uint32_t layer: 3;
+        uint32_t roc : 3;
+        uint32_t rob: 3;       
+        uint32_t mcm : 4;     
+        uint32_t cpu : 2;    
+      }__attribute__((__packed__));
+    };
+};
+struct trdTestPattern6{
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  //             1zzzz  pppppppp        pppppppp1
+  union {
+    uint32_t word;  //HC header0
+      struct {
+        uint32_t eventcount;  // lower 4 bits of e counter.
+        uint32_t stack: 5;   // starting at 1
+        uint32_t layer: 3;  
+        uint32_t roc : 3; 
+        uint32_t rob : 3;      
+        uint32_t mcm : 4;       
+        uint32_t cpu : 2;        
+        uint32_t counter : 6;     
+        uint32_t oddadc : 2;       
+    }__attribute__((__packed__));
+  };
+};
+
+struct DigitMCMData {
+  //             10987654321098765432109876543210
+  // uint32_t:   00000000000000000000000000000000
+  union {
+      uint32_t word0;
+      struct{
+        uint32_t a : 2;
+        uint32_t b: 5;
+        uint32_t adc : 21;       //pad plane
```

Comment:
The comment 'pad plane' is misleading...

Reply:
thats a mis copy.